### PR TITLE
Automated location and Filtering

### PR DESCRIPTION
**Pull request by YvonneAriri**
Description:

- The server-side code has been updated to filter events based on the user's username using the useParams hook.
- An "Ongoing" section has been added to the application, enabling users to see events that have already started but are still ongoing, even if their start time is in the past.
- The Google Maps API has been integrated into the application to implement automated location input

Milestone:
The milestones achieved in this implementation include:

- Server-side filtering: The ability to filter events on the server side using the useParams hook.
- Ongoing section: Implementing a section that displays events that have already started but are still ongoing.
- Google Maps integration: Incorporating the Google Maps API to facilitate automated location input.

Test Plan:
Eveprep10

### CODE REVIEW

File: event-app-ui/index.html [hunk 1]
```diff
@@ -9,5 +9,15 @@
...
+    <script type="module">
+      import { API_KEY } from "./api-key.js";
+
+      // Load google maps javascript API
+      var apiScript = document.createElement("script");
+      apiScript.src = `https://maps.googleapis.com/maps/api/js?key=${API_KEY}&libraries=places&callback=Function.prototype`;
+      apiScript.innerHTML = null;
+      apiScript.async = true;
+      document.body.appendChild(apiScript);
+    </script>
```
Comment (sophia129) [inline event-app-ui/index.html:21]: Why is this code not in the main.jsx file?

File: event-app-ui/src/components/EventSection/EventSection.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+              <p>
+                {" "}
+                {new Date(value.starttime)
+                  .toLocaleDateString("en-US", {
+                    year: "2-digit",
+                    month: "2-digit",
+                    day: "2-digit",
+                    hour: "numeric",
+                    minute: "numeric",
+                    hour12: true,
```
Comment (sophia129) [inline event-app-ui/src/components/EventSection/EventSection.jsx:35]: this config object can be a variable and reused for the end time block below

File: event-app-ui/src/components/Profile/Profile.jsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import Popup from "components/Popup/Popup";
 import Navbar from "components/Navbar/Navbar";
 import EditProfile from "../EditProfile/EditProfile";
 import { FaUserCircle } from "react-icons/fa";
+import EventSection from "../EventSection/EventSection";
```
Comment (sophia129) [inline event-app-ui/src/components/Profile/Profile.jsx:9]: make sure you're minimizing relative imports in all components

**Review by sophia129 — APPROVED** [review]

File: event-app-ui/index.html (hunk 1 above)
Comment (YvonneAriri) [reply event-app-ui/index.html:21]: I am using the inline API loading in the Documentation . 
This is the link https://developers.google.com/maps/documentation/javascript/load-maps-js-api#use-legacy-tag

File: event-app-ui/src/components/Profile/Profile.jsx (hunk 3 above)
Comment (YvonneAriri) [reply event-app-ui/src/components/Profile/Profile.jsx:9]: I will

**Pull request merged**